### PR TITLE
display progress counters at fixed locations

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes https://github.com/microsoft/vcpkg-tool/pull/437#issuecomment-1093889391

cc @dg0yt

### CODE REVIEW

File: include/vcpkg/base/messages.h [hunk 1]
```diff
@@ -273,6 +273,8 @@ namespace vcpkg::msg
     DECLARE_MSG_ARG(vcpkg_line_info, "/a/b/foo.cpp(13)");
     DECLARE_MSG_ARG(vendor, "Azure");
     DECLARE_MSG_ARG(version, "1.3.8");
+    DECLARE_MSG_ARG(action_index, "340");
+    DECLARE_MSG_ARG(action_count, "1400");
```
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:277]: I think one of these can just reuse msg::count.

File: include/vcpkg/remove.h [hunk 2]
```diff
@@ -16,7 +16,10 @@ namespace vcpkg::Remove
         YES
     };
 
-    DECLARE_MESSAGE(RemovingPackage, (msg::spec), "", "Removing {spec}");
+    DECLARE_MESSAGE(RemovingPackage,
+                    (msg::action_index, msg::action_count, msg::spec),
+                    "",
+                    "Removing {action_index}/{action_count} {spec}");
```
Comment (BillyONeal) [inline include/vcpkg/remove.h:22]:
```suggestion
                    "[{action_index}/{action_count}] Removing {spec}");
```

I think this was dg0yt's suggestion? (same for other examples)

I don't have strong opinions here I just want to make sure you considered this.

**Review by BillyONeal — COMMENTED** [review]
I'm happy with this but want to confirm you considered the [x/y] prefix and discarded it for some reason?

File: include/vcpkg/remove.h (hunk 2 above)
Comment (autoantwort) [reply include/vcpkg/remove.h:22]: Imo [x/y] looks only good when every line is prefixed with [x/y]

**Review by BillyONeal — APPROVED** [review]

**Review by vicroms — APPROVED** [review]
Personally, I'd prefer the `[x/y] ...` format but I'm not blocking over that.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**